### ISSUE TITLE
Spike: CAS SSO reorientation

### ISSUE THREAD

**Issue by jillpe** [notch8/utk-hyku#410]
# Summary
ref: #2 
We need to reorient ourselves with where we are at in the process, so we can break down the remaining tasks. If there are questions we need answers to to unblock the task breakdown, we need to take note of them in a follow up spike ticket and tag it in the [meeting notes](https://docs.google.com/document/d/1GsoeK7gkBq3yziAVIK4Fijihd7jNo5nkc0Fnj8K1-B4/edit#)

# Acceptance Criteria
- [x] Current status of CAS SSO is documented
- [x] Relevant prs, tickets, notes, etc. are documented 
- [x] Determining how much is left and if additional breakdown is needed

# TODO: 
- [ ] https://github.com/scientist-softserv/utk-hyku/issues/368 **(To be done by Team Teal 6)**
- [x] https://github.com/scientist-softserv/utk-hyku/issues/2
- [x] https://github.com/scientist-softserv/utk-hyku/issues/207

## Notes (from Crystal on next steps)

- [ ]     Install gem (Omniauth) https://github.com/scientist-softserv/utk-hyku/issues/2
- [ ] From UTK: a metadata profile for their SSO fields; contact from their side for SSO (want to be able to work directly with them)

  - [ ]     Server to connect to; username, id, email
  - [ ]     Mapping from theirs to our (getting theirs first is easier)

- [ ] Need an ssl cert and domain name (formal domain name)

  - [ ]     domain name: digitalcollections.lib.utk.edu
  - [ ]     Need to set up formal domain (https://github.com/scientist-softserv/utk-hyku/issues/368) - TEAL TEAM 6 WILL HANDLE THIS
    - [ ]         It can change later (changing means you need to change SSO though)
  - [ ]     SSL cert (we typically handle it) - TEAL TEAM 6 WILL HANDLE THIS
  - [ ]         Once domain is set up, we create this and will send them the details and they will put in to their system
   - [ ]         Similar to SSH keys
- [ ]     Include Rob on the domain name set up

- [ ] Ask if they have documentation on SSO
- [ ] get sso accounts for the team v

**Comment by ShanaLMoore** [comment]
From my understanding, this process shouldn't start until the client has everything set up on their end. Such as a domain+. cc @crisr15 may be more familiar with this process and its requirements. And I believe currently only Rob has experience with setting this up.

**Comment by jillpe** [comment]
Notes on steps for this (from Crystal):
* Steps
    * Install gem (Omniauth) https://github.com/scientist-softserv/utk-hyku/issues/2
    * From UTK: a metadata profile for their SSO fields; contact from their side for SSO (want to be able to work directly with them)
        * Server to connect to; username, id, email
        * Mapping from theirs to our (getting theirs first is easier)
    * Need an ssl cert and domain name (formal domain name) 
        * domain name: digitalcollections.lib.utk.edu
        * Need to set up formal domain ([another ticket](https://github.com/scientist-softserv/utk-hyku/issues/368)) -  **TEAL TEAM 6 WILL HANDLE THIS**
            * It can change later (changing means you need to change SSO though)
        * SSL cert (we typically handle it) - **TEAL TEAM 6 WILL HANDLE THIS**
            * Once domain is set up, we create this and will send them the details and they will put in to their system
            * Similar to SSH keys
        * Include Rob on the domain name set up 
    * Ask if they have documentation on SSO

**Comment by jillpe** [comment]
2 gem options:

- [Omniauth](https://github.com/dlindahl/omniauth-cas): This is the preferred one (it is also compatible with other forms of SSO)
 - [Devise Auth](https://github.com/nbudin/devise_cas_authenticatable)


Potentially Helpful PRs
https://gitlab.com/notch8/dot/-/issues/5
https://gitlab.com/notch8/dot/-/issues/4
https://gitlab.com/notch8/wustl-hyrax/-/issues/12
https://playbook-staging.notch8.com/en/samvera/SSO/saml

**Comment by ShanaLMoore** [comment]
ESSI uses [omniauth-cas](https://github.com/IU-Libraries-Joint-Development/essi/commit/c9ea9a68b1013f22b442d2c814802e0662772052)
